pkg/restapi/device/deviceapi.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package device
+
+import (
+	"crypto/rand"
+	"encoding/binary"
+
+	"github.com/duo-labs/webauthn/protocol"
+	"github.com/duo-labs/webauthn/webauthn"
+	"github.com/trustbloc/edge-agent/pkg/restapi/common/store/user"
+)
+
+// Device represents the user model
+type Device struct {
+	id          uint64
+	name        string
+	displayName string
+	credentials []webauthn.Credential
+}
+
+// NewDevice creates and returns a new Device
+func NewDevice(user *user.User) *Device {
+	device := &Device{}
+	device.id = randomUint64()
+	device.name = user.Name
+	device.displayName =user.FamilyName
+	device.credentials = []webauthn.Credential{}
+
+	return device
+}
+
+func randomUint64() uint64 {
+	buf := make([]byte, 8)
+	rand.Read(buf)
+	return binary.LittleEndian.Uint64(buf)
+}
+
+// WebAuthnID returns the user's ID
+func (d Device) WebAuthnID() []byte {
```

Comment:
This identifier should have a clear relationship to the unique identifier currently assigned to users in our model (`User.Sub`).

Please set it to `User.Sub` for now.

Reply:
done

Reply:
done